Uninitialized variable warnings

Xeverous opened the pull request.

Xeverous commented:
Initialization in C++ is a mess and I really don't know which approach to use in the case where there is a `struct foo { int x = 0; bar y; };` where `bar` actually has a default ctor.

- write `foo{ 1 }` - `y` is not initialized even though `bar` has a ctor (aggregate rules)
- write `foo(1)` - `foo` needs an explicitly written ctor

omartijn commented:
I think the safest option is to add an initializer to all struct variables, so your example struct would be something like this:

```
struct foo {
    int x{};
    bar y{};
}
```

This should ensure that all members are always initialized. If you explicitly construct them, e.g. `foo{ 1 }` will initialize `x` to `1` instead of the default `0`.

Xeverous commented:
Makes sense. I will use `{}` then, check for warnings and rebase the PR because it got somewhat stale and got some conflicts.

Xeverous commented on lib/src/element/gallery/menu.cpp:
```diff
@@ -141,24 +141,24 @@ namespace cycfi { namespace elements
 
       if (mod & mod_alt)
-         r1 += u8"⌥";
+         r1 += u8"âŚĄ";
```
Actually, that's my fault. These changes should not be present. I opened the file and the editor was set to something different than UTF-8 which caused some characters to be interpreted as invalid, hence the difference in saved characters. I will remove such changes from the PR.

Xeverous commented:
Rebased and updated the commit with what @omartijn proposed. The only warnings left are the ones from `stb_image.h` and 1 from `json/json.hpp`.

@djowel For stb, it's a separate library, just pasted into this repo. Not sure if we should modify it.
